controls/os_spec.rb

```diff
@@ -282,3 +290,99 @@
     end
   end
 end
+
+control 'os-14' do
+  impact 1.0
+  title 'All password change dates are in the past'
+  desc 'The password change date is used to detect expired passwords. Entering future dates might circumvent that.'
+  describe shadow.where { last_change.to_i > (Date.today - Date.new(1970, 1, 1)).to_i } do
+    its('users') { should be_empty }
+  end
+end
+
+control 'os-15' do
+  impact 1.0
+  title 'All system users are non-login'
+  desc 'The login of system users should be disabled'
+  system_users.each do |user|
+    next if system_users_non_login_permitlist.include? user['user']
+
+    describe OpenStruct.new(user) do
+      its('shell') { should be_in ['/bin/false', '/sbin/nologin', '/usr/bin/false', '/usr/sbin/nologin'] }
+    end
+  end
+end
+
+control 'os-16' do
+  impact 1.0
+  title 'User \'root\' should be member of group \'root\' with gid \'0\''
+  desc 'This prevents root-owned files and directories to be accessible to non-privileged users'
+  describe passwd.uids(0) do
+    its('users') { should cmp 'root' }
+    its('gids') { should cmp 0 }
+  end
+  describe etc_group.where(gid: 0) do
+    its('groups') { should cmp 'root' }
+    its('users') { should be_empty }
+  end
+end
+
+control 'os-17' do
+  impact 1.0
+  title 'Prevent + or - fields in passwd an related files used by NIS'
+  desc 'NIS is insecure and should not be used'
```

Comment:
A reference to the rule and reasoning for be beneficial

Reply:
This rule is also based on CIS, but expanded upon. Cis only mentions that passwd entries should not include `+`, but NIS allow [allows](https://tldp.org/LDP/nag2/x-087-2-nis.passwd.html) `-` entries. This rule tests for both. So I don't have any reference to the exact rule. I could drop this rule as well if required.